### PR TITLE
feat: add getCitiesByRegionId & getDefaultRegionsByCountryIdApi & getRegionsByCountryIdApi & common address types & handleOnAddressValueChange & region combobox & translation

### PR DESCRIPTION
**Pull request by ahmeterenodaci**
## Description

Create a selectedFields variable, give values if default value exists.
resim

On auto form values changed, call handleOnAddressValueChange with required props. This will handle all region/city requests.
resim

When user submits form, just override selectedFields
resim

## How has this been tested?

- [ ] Prod
- [ ] Development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/web/src/app/[lang]/app/actions/LocationService/types.ts
Comment (ertugrulcan-ays) [inline apps/web/src/app/[lang]/app/actions/LocationService/types.ts:1]: I am not sure about those exports

**Review by ertugrulcan-ays — DISMISSED** [review]
LGTM

File: apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx [hunk 1]
```diff
@@ -71,6 +91,81 @@ export function getAddressFieldConfig(params: {
...
+    setSelectedFields((current) => ({
+      ...current,
+      regionId,
+      cityId: "",
+    }));
+    const cities = await getCitiesByRegionId({ regionId });
+    if (cities.type === "success") {
+      setCityList(cities.data);
+      return;
+    }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx:122]: did we handle error case here?

File: apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx [hunk 2]
```diff
@@ -71,6 +91,81 @@ export function getAddressFieldConfig(params: {
...
+    );
+    if (selectedCountry?.hasRegion) {
+      const regions = await getRegionsByCountryIdApi({
+        countryId,
+      });
+      setCityList([]);
+      if (regions.type === "success") {
+        setRegionList(regions.data);
+        return;
+      }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx:147]: what if we got an error ?

File: apps/web/src/app/[lang]/app/actions/LocationService/types.ts
Comment (ahmeterenodaci) [reply apps/web/src/app/[lang]/app/actions/LocationService/types.ts:1]: why?

File: apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx [hunk 3]
```diff
@@ -71,6 +91,81 @@ export function getAddressFieldConfig(params: {
...
+      const regionId = regions.data;
+      setRegionList([]);
+      setSelectedFields((current) => ({
+        ...current,
+        regionId,
+      }));
+      void getCities(regionId);
+      return;
+    }
+    setCityList([]);
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/actions/LocationService/schemas.tsx:165]: we did not consider errors here as well.

**Review by a0m0rajab — DISMISSED** [review]
LGTM 👍

**Review by a0m0rajab — APPROVED** [review]

**Review by ertugrulcan-ays — APPROVED** [review]

**Pull request merged**